super-ddf.c

```diff
@@ -975,30 +979,31 @@ static int load_ddf_headers(int fd, struct ddf_super *super, char *devname)
 		return 2;
 	}
 	super->active = NULL;
-	if (load_ddf_header(fd, be64_to_cpu(super->anchor.primary_lba),
-			    dsize >> 9,  1,
-			    &super->primary, &super->anchor) == 0) {
-		if (devname)
-			pr_err("Failed to load primary DDF header on %s\n", devname);
-	} else
+	if (load_ddf_header(fd,
+			    be64_to_cpu(super->anchor.primary_lba),
+			    dsize >> 9,
+			    DDF_HEADER_PRIMARY,
+			    &super->primary,
+			    &super->anchor)) {
 		super->active = &super->primary;
+	} else if (devname) {
+		pr_err("Failed to load primary DDF header on %s\n", devname);
+	}
 
-	if (load_ddf_header(fd, be64_to_cpu(super->anchor.secondary_lba),
-			    dsize >> 9,  2,
-			    &super->secondary, &super->anchor)) {
-		if (super->active == NULL
```

Comment:
Hey... did we need this code that was executed if we successfully loaded the primary header?  It looked like it took the secondary if it was newer?

Reply:
ok sure. I removed the cases where `openflag` would ever be set, because we fail the load now if `openflag` is ever set. But I'll add back the sequence check if both metadatas are valid. Should never be possible, the way it is written, but doesn't hurt.